### PR TITLE
WIP : Changing from Intake to Piston

### PR DESCRIPTION
**Pull request by KingOfSquares**
Hello. [Link to Piston](https://github.com/EngineHub/Piston)

I was inspired by #210 to start a migration to a newer more supported framework for in-game commands. 

While this is WIP formatting is probably going to be pretty ruined.

Things that needs to be done for this to work:

- [x] Change the annotation parts in the XXXcommands.java
- [x] Change the Exceptions in the XXXcommands.java
- [x] Add a replacement for @Default
- [ ] Add permissions (Probably with CommandConditionGenerator [link](https://github.com/EngineHub/Piston/blob/master/core-ap/runtime/src/main/java/org/enginehub/piston/gen/CommandConditionGenerator.java)
- [ ] Add the implementation part
- [ ] Remove all other dependencies on Intake (Not sure how much there is of this..)

Other things

- [ ] Look into moving translation of exception out of the commands.java

All feedback appreciated!! :)

Signed-off-by: KingOfSquares <simonmorland@gmail.com>

### CODE REVIEW

**Comment by Electroid** [comment]
I like where this is going, keep going. I'd rather not *also* maintain Intake.

**Comment by AustinLMayes** [comment]
@KingOfSquares While converting, it would be nice if we could implement a simple translatable command backbone - which shifts all translation/rendering to the main `onCommand` entrypoint in `PGM`. 

See the [WorldEdit Implementation](https://github.com/EngineHub/WorldEdit/blob/master/worldedit-core/src/main/java/com/sk89q/worldedit/extension/platform/PlatformCommandManager.java) for an example. **Note:** That file is quite long and does a lot of extra work to support WE's multiple platforms, the main thing I"m referring to is the `catch` blocks in the `handle` method. 

There are a lot of instances in code currently where exceptions are either in plain English or translated in-method, which isn't great.

**Comment by KingOfSquares** [comment]
Will try that @AustinLMayes !

**Comment by KingOfSquares** [comment]
Im having issues with conflicting Guava code. So until i can resolve that this is halted.

**Comment by TheMolkaPL** [comment]
> issues with conflicting Guava code

Can you explain more?

**Comment by KingOfSquares** [comment]
> > issues with conflicting Guava code
> 
> Can you explain more?

The Minecraft Server uses Guava 17 for 1.8, but i need Guava > 20.00 for Piston to work. Im trying to get piston to relocate its own guava files so it wont conflict but.. i'm not able to do that yet. Posted about it in #debug on OCC Discord

**Comment by KingOfSquares** [comment]
Right now this seems like a project a little bit out of reach for me. If someone manages to get Piston to work i would love to help, but i am going to try and work on other issues as of now.
